packages/discord.js/src/structures/Message.js

```diff
@@ -138,9 +138,9 @@ class Message extends Base {
     if ('components' in data) {
       /**
        * A list of MessageActionRows in the message
-       * @type {MessageActionRow[]}
+       * @type {ActionRow[]}
        */
-      this.components = data.components.map(c => BaseMessageComponent.create(c, this.client));
+      this.components = data.components.map(createComponent);
```

Comment:
From what I heard this is much more performant than passing the method directly

```suggestion
      this.components = data.components.map(c => createComponent(c));
```